lib/compilers/d8.ts

```diff
@@ -0,0 +1,341 @@
+// Copyright (c) 2023, Compiler Explorer Authors
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+//     * Redistributions of source code must retain the above copyright notice,
+//       this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+import path from 'path';
+
+import fs from 'fs-extra';
+import _ from 'underscore';
+
+import {logger} from '../logger.js';
+
+import {BaseCompiler} from '../base-compiler.js';
+import {JavaCompiler} from './java.js';
+import {KotlinCompiler} from './kotlin.js';
+
+import type {PreliminaryCompilerInfo} from '../../types/compiler.interfaces.js';
+import type {ParsedAsmResult, ParsedAsmResultLine} from '../../types/asmresult/asmresult.interfaces.js';
+import {CompilationResult, ExecutionOptions} from '../../types/compilation/compilation.interfaces.js';
+import type {ParseFiltersAndOutputOptions} from '../../types/features/filters.interfaces.js';
+
+export class D8Compiler extends BaseCompiler {
+    static get key() {
+        return 'android-d8';
+    }
+
+    lineNumberRegex: RegExp;
+    methodEndRegex: RegExp;
+
+    javaPath: string;
+    javacPath: string;
+    kotlincPath: string;
+
+    constructor(compilerInfo: PreliminaryCompilerInfo, env) {
+        super({...compilerInfo}, env);
+
+        this.lineNumberRegex = /^\s+\.line\s+(\d+).*$/;
+        this.methodEndRegex = /^\s*\.end\smethod.*$/;
+
+        // These are the paths of the java/javac and kotlinc versions that are
+        // currently being installed as a result of java 16.0.1 and kotlin
+        // 1.9.20 being marked as dependencies of android-d8 in the infra repo.
+        this.javaPath = '/opt/compiler-explorer/jdk-16.0.1/bin/java';
```

Comment:
Please don't hardcode any of these. Please make them configuration options for the compiler so we can set them in the configuration files, and folks can configure them for their local setups accordingly.  See, for example the code in `kotlin.ts` that sets up its javahome according to the configuration `java_home`.

Reply:
Done! Please let me know if something doesn't look right in the changes to `android-java.amazon.properties` and `android-kotlin.amazon.properties`; I understood the kotlin.ts java_home example to mean that the paths should be added only to the `amazon` prop file.